src/test/java/com/ipiecoles/java/java350/service/EmployeServiceTest.java

```diff
@@ -0,0 +1,265 @@
+
+import com.ipiecoles.java.java350.exception.EmployeException;
+import com.ipiecoles.java.java350.model.Employe;
+import com.ipiecoles.java.java350.model.Entreprise;
+import com.ipiecoles.java.java350.model.NiveauEtude;
+import com.ipiecoles.java.java350.model.Poste;
+import com.ipiecoles.java.java350.repository.EmployeRepository;
+
+import org.assertj.core.api.Assertions;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.persistence.EntityExistsException;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+@ExtendWith(MockitoExtension.class)
+
+ class EmployeServiceTest {
+
+    @InjectMocks
+    EmployeService employeService;
+
+    @Mock
+    EmployeRepository employeRepository;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this.getClass());
+    }
+@ParameterizedTest
+@CsvSource({"C00001,1500,1000,1,6", "C00001, 800, 1000, 5,4", "C00001, 1055, 1000, 5,7", "C00001, 1000, 1000,5,6" })
+ void testCalculPerformanceCommercial(String matricule, Long caTraite, Long objectifCa, Integer perfBase, Integer perfAttendu) throws EmployeException {
+    //Given
+    Employe employe = new Employe("Marc", "Davis", matricule, LocalDate.now(), Entreprise.SALAIRE_BASE, perfBase, 1.0);
+    when(employeRepository.findByMatricule(matricule)).thenReturn(employe);
+
+    //When
+    employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa);
+    ArgumentCaptor<Employe> employeArgumentCaptor = ArgumentCaptor.forClass(Employe.class);
+    verify(employeRepository, Mockito.times(1)).save(employeArgumentCaptor.capture());
+
+    //Then
+    Assertions.assertThat(employeArgumentCaptor.getValue().getPerformance()).isEqualTo(perfAttendu);
+}
+    @Test
+
+     void testPerformanceCommercialcaTraiteNull() throws EmployeException {
+//given
+        Employe employe = new Employe("Marc", "Davis", "C00001", LocalDate.now(), Entreprise.SALAIRE_BASE,Entreprise.PERFORMANCE_BASE,1.0);
+        //Mockito.when(employeRepository.findByMatricule("C00001")).thenReturn(employe);
+        //when
+        //employeService.calculPerformanceCommercial("C00001",null, 1000L);
+        //ArgumentCaptor<Employe> EmpPerf = ArgumentCaptor.forClass(Employe.class);
+        //Mockito.verify(employeRepository, Mockito.times(1)).save(EmpPerf.capture());
+        //then
+
+        try{
+            employeService.calculPerformanceCommercial("C00001",null, 1000L);
+
```

Comment:
N'oublie pas le `fail` pour le cas ou l'exception ne serait pas lancée
```suggestion
            Assertions.fail("Aurait du planter !");
```